src/py21cmfast/cli.py

```diff
@@ -1,827 +1,733 @@
+from rich.console import Console, group
+from rich.logging import RichHandler
+from rich.panel import Panel
+from rich.rule import Rule
+from rich.text import Text
+
+from . import __version__, plotting
+from .drivers import coeval as cvlmodule
+from .drivers.coeval import generate_coeval
 from .drivers.lightcone import run_lightcone
-from .drivers.single_field import (
-    compute_initial_conditions,
-    compute_ionization_field,
-    compute_spin_temperature,
-    perturb_field,
-)
-from .io.caching import OutputCache
+from .drivers.single_field import compute_initial_conditions
+from .io.caching import CacheConfig, OutputCache, RunCache
 from .lightconers import RectilinearLightconer
-from .run_templates import create_params_from_template, load_template_file
-from .wrapper._utils import camel_to_snake
+from .run_templates import TOMLMode, _get_inputs_as_dict, list_templates, write_template
+from .wrapper._utils import snake_to_camel
 from .wrapper.inputs import (
+    AstroOptions,
+    AstroParams,
+    CosmoParams,
     InputParameters,
-    get_logspaced_redshifts,
-)
-from .wrapper.outputs import (
-    HaloBox,
-    InitialConditions,
-    IonizedBox,
-    PerturbedField,
-    TsBox,
-    XraySourceBox,
+    MatterOptions,
+    SimulationOptions,
 )
 
+cns = Console()
 
-def _ctx_to_dct(args):
-    dct = {}
-    j = 0
-    while j < len(args):
-        arg = args[j]
-        if "=" in arg:
-            a = arg.split("=")
-            dct[a[0].replace("--", "")] = a[-1]
-            j += 1
-        else:
-            dct[arg.replace("--", "")] = args[j + 1]
-            j += 2
-
-    return dct
-
-
-def _update(obj, ctx):
-    """Override attributes of an object using the click context."""
-    # Try to use the extra arguments as an override of config.
-    kk = list(ctx.keys())
-    for k in kk:
-        # noinspection PyProtectedMember
-        if hasattr(obj, k):
-            try:
-                val = getattr(obj, f"_{k}")
-                setattr(obj, f"_{k}", type(val)(ctx[k]))
-                ctx.pop(k)
-            except (AttributeError, TypeError):
-                with contextlib.suppress(AttributeError):
-                    val = getattr(obj, k)
-                    setattr(obj, k, type(val)(ctx[k]))
-                    ctx.pop(k)
-
-
-def _get_inputs_from_ctx(ctx, seed, config="simple", **kwargs):
-    """Construct InputStruct objects from the click context and config."""
-    # these functions use ctx and kwargs to override params set in config,
-    # where config is either a native template string or path to TOML file
-    ctx = _ctx_to_dct(ctx.args)
-    param_structs = create_params_from_template(config, **ctx)
-    inputs = InputParameters(random_seed=seed, **{**param_structs, **kwargs})
+FORMAT = "%(message)s"
+logging.basicConfig(
+    level="WARNING",
+    format=FORMAT,
+    datefmt="[%X]",
+    handlers=[RichHandler(rich_tracebacks=True, console=cns)],
+)
+logger = logging.getLogger("py21cmfast")
```

Comment:
I'm guessing we don't want to use `__name__` here since this is a special logger for the CLI?

Reply:
I think if we use `__name__`, it will affect the logger called `"py21cmfast.cli"` and all its children, but not the loggers from e.g. `py21cmfast.wrapper` etc. So yeah, I wanted to name it with the top-level 21cmFAST package.